src/main/java/org/example/StudentDAO.java

```diff
@@ -0,0 +1,51 @@
+package org.example;
+
+import org.example.entities.Student;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import java.util.List;
+
+public class StudentDAO {
+    public void addStudent(Student student) {
+        SessionFactory sessionFactory = HibernateSession.getSessionFactory();
+        try (Session session = sessionFactory.openSession()) {
+            Transaction transaction = session.beginTransaction();
+            session.save(student);
+            transaction.commit();
+        }
+    }
+
+    public void updateStudent(Student student) {

```

Comment:
You need to make sure that the student has an id assigned, otherwise could you please tell me, what will Hibernate do if the id is null?